python/tank/platform/engine.py

```diff
@@ -1275,14 +1265,41 @@ def _do_invoke(self):
                         """
                         self._res = self._fn()
 
-                # Make sure that the invoker exists in the main thread:
-                invoker = Invoker()
-                if QtGui.QApplication.instance():
-                    invoker.moveToThread(QtGui.QApplication.instance().thread())
-                return invoker
+                class AsyncInvoker(QtCore.QObject):
```

Comment:
That is why we must have introduced these within the method originally.

I think a better refactor for these would be to completely abstract out what the invoker is. So there wouldn't be any of this testing for Qt and QApp nonsense in multiple places.